### PR TITLE
[JENKINS-54889] Avoid NPE with graphlistener (null check on value before adding to the map)

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: olivier lamy <olamy@apache.org>

# Description

See [JENKINS-54889](https://issues.jenkins-ci.org/browse/JENKINS-54889).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by imeredith** [comment]
should there be a try catch around whole method? incase there is ever any other issues?

**Comment by imeredith** [comment]
nvm i see cps pr, :bee:

**Comment by olamy** [comment]
IMHO nope.
Reading javadoc: https://docs.oracle.com/javase/10/docs/api/java/util/concurrent/ConcurrentHashMap.html#put(K,V)
null check on values/keys should be enough. (it's not javascript code so everything is under control and documented :P )

File: blueocean-events/src/main/java/io/jenkins/blueocean/events/PipelineEventListener.java [hunk 1]
```diff
@@ -71,8 +71,12 @@ public void onNewHead(FlowNode flowNode) {
         // test whether we have a stage node
         if (PipelineNodeUtil.isStage(flowNode)) {
             List<String> branch = getBranch(flowNode);
-            currentStageName.put(flowNode.getExecution(), flowNode.getDisplayName());
-            currentStageId.put(flowNode.getExecution(), flowNode.getId());
+            if(flowNode.getExecution()!=null && flowNode.getDisplayName()!=null) {
```
Comment (kzantow) [inline blueocean-events/src/main/java/io/jenkins/blueocean/events/PipelineEventListener.java]: My only question about this is what scenarios is `flowNode.getExecution()` ever `null`? Just curious if it's indicative of a bigger/different problem.

Comment (olamy) [reply blueocean-events/src/main/java/io/jenkins/blueocean/events/PipelineEventListener.java]: well sure flowNode.getExecution() shouldn't be null. I will remove but this doesn't hurt neither have a big impact....

**Pull request merged**